# Project Part #4 : CPSMA 3913 Dr. Jacob's Discrete Mathematics
## Counting
### Trevin Madron - 237379

As with my last example, I have 6 different colored highlighters in my bag for college. The colors are yellow, blue, pink, orange, green, and purple. To extend upon this however, I also have two different colored pens, one black and one red. I also have a pencil, and a permanent marker. For this example I'll be looking at different probabilities as I pull out two random items on each attempt. Aside from the fact that this is making me feel like I have way too many writing utensils stuffed in my bag, lets get down to it. Here we find the factorial of 10-- how many different sequences can I choose all 10 items? 

In [1]:
import math

math.factorial(10)

3628800

How many different outcomes could we have if order mattered, but we replaced the item after each pick? To get that answer, we would just multiply the amount of choices for the first pick (10) and the amount of choices for the second pick (also 10, since we replaced the first pick). We would get 
$$10 * 10 = 10^2$$

In [2]:
10**2

100

What if order didn't matter but we stopped replacing items? This is known as a combination and we can use 10 choose 2, which is the amount of ways we can choose 2 items out of a set of 10.
$$\frac{x!}{(n-k)!k!}$$

In [3]:
math.factorial(10) / ((math.factorial(10 - 2)) * math.factorial(2))

45.0

Flipping around, what if order DID matter and we didn't replace any items? This is called a partial permutation.
$$\frac{x!}{(n-k)!}$$

In [4]:
math.factorial(10)/(math.factorial(10 - 2))

90.0

In esssense, a combination refers to the scenario in which order doesn't matter. Permutations however, involve scenarios in which order DOES matter.

Now to find the coeffecient of the 5th order of the following:

$$(x - 3)^9 $$

For now, I'm going to substitute the 3 for y. First lets get 9 choose 5. 

In [5]:
math.factorial(9) / ((math.factorial(9 - 5)) * math.factorial(5))

126.0

$$(x^9)(-y^0), (x^8)(-y^1), (x^7)(-y^2), (x^6)(-y^3), (x^5)(-y^4)$$

Since we are trying to find the 5th order, I displayed the first 5 in the sequence. The first variable (x) starts at the top (in our case, 9 as the exponent given in the starting equation) and decreases to zero. The second variable (y) starts at zero and increases to 9. We need the 5th one in this list, which would be:

$$(x^5)(-y^4)$$

Then we will append the output from 9 choose 5. You could also use a Pascal's Triangle as well to get the same result. While we're at it, we will replace the y with its original value, 3.

$$126(x^5)(-y^4) = 126(x^5)(-3^4) = 126(x^5)(81) = 81 * 126(x^5) = 10206(x^5)$$

The coefficient of the 5th order for this equation is 10206.